### PR TITLE
Support deployment in Public cloud

### PR DESCRIPTION
**Pull request by outscale-mdr**
- all nodes have public Ip
- loadbalancer for the kube-apiserver

(See https://github.com/outscale-mdr/osc-k8s-rke-cluster/actions/runs/2745246478 for run proof)

### CODE REVIEW

File: control-planes.tf [hunk 1]
```diff
@@ -20,9 +20,19 @@ resource "outscale_keypair" "control-planes" {
...
+resource "outscale_public_ip_link" "control-planes" {
+  count     = var.public_cloud ? var.control_plane_count : 0
+  vm_id     = outscale_vm.control-planes[count.index].vm_id
+  public_ip = outscale_public_ip.control-planes[count.index].public_ip
+}
+
 resource "outscale_security_group" "control-plane" {
   description = "Kubernetes control-planes (${var.cluster_name})"
-  net_id      = outscale_net.net.net_id
+  net_id      = var.public_cloud ? "" : outscale_net.net[0].net_id
```
Comment (outscale-vbr) [inline control-planes.tf:35]: why sometimes var.public_cloud ? "" and other it is var.public_cloud ? nil for net_id

**Review by outscale-vbr — APPROVED** [review]

File: docs/README.md [hunk 2]
```diff
@@ -3,14 +3,19 @@
...
 The Kubernetes cluster is deployed inside a [Net](https://wiki.outscale.net/display/EN/About+VPCs) with two [Subnets](https://wiki.outscale.net/display/EN/Getting+Information+About+Your+Subnets):
 - One subnet (10.0.0.0/24) containing:
   - A bastion host
   - A [NAT Service](https://wiki.outscale.net/display/EN/About+NAT+Gateways) to provide internet access to nodes.
   - A load balancer for kube-apiserver
 - One subnet (10.0.0.1/24) containing all nodes (control plane and worker nodes)
 
+## Public
+The Kubernetes cluster is deployed in the  public cloud :
+- All nodes have a public IP:
```
Comment (outscale-vbr) [inline docs/README.md:16]: Remove :

File: control-planes.tf (hunk 1 above)
Comment (jerome-jutteau) [inline control-planes.tf:35]: "" -> null

**Review by jerome-jutteau — CHANGES_REQUESTED** [review]

**Comment by outscale-mdr** [comment]
See success of update https://github.com/outscale-mdr/osc-k8s-rke-cluster/actions/runs/2977682512

File: workers.tf [hunk 3]
```diff
@@ -23,21 +23,32 @@ resource "outscale_keypair" "workers" {
...
+  count = var.public_cloud ? var.worker_count : 0
+}
+
+resource "outscale_public_ip_link" "workers" {
+  count     = var.public_cloud ? var.worker_count : 0
+  vm_id     = outscale_vm.workers[count.index].vm_id
+  public_ip = outscale_public_ip.workers[count.index].public_ip
+}
+
+
```
Comment (jerome-jutteau) [inline workers.tf:39]: In public cloud, workers will receive a publique IP even if you don't attach an EIP unless you specify `private_only` tag.
https://docs.outscale.com/en/userguide/About-Instances.html

**Review by jerome-jutteau — CHANGES_REQUESTED** [review]

**Pull request merged**